API-Key key masking may cause StringIndexOutOfBoundsException

```
Caused by: java.lang.StringIndexOutOfBoundsException: Range [0, 5) out of bounds for length 4
at java.base/jdk.internal.util.Preconditions$1.apply(Preconditions.java:55)
at java.base/jdk.internal.util.Preconditions$1.apply(Preconditions.java:52)
at java.base/jdk.internal.util.Preconditions$4.apply(Preconditions.java:213)
at java.base/jdk.internal.util.Preconditions$4.apply(Preconditions.java:210)
at java.base/jdk.internal.util.Preconditions.outOfBounds(Preconditions.java:98)
at java.base/jdk.internal.util.Preconditions.outOfBoundsCheckFromToIndex(Preconditions.java:112)
at java.base/jdk.internal.util.Preconditions.checkFromToIndex(Preconditions.java:349)
at java.base/java.lang.String.checkBoundsBeginEnd(String.java:4861)
at java.base/java.lang.String.substring(String.java:2830)
at io.github.jeremylong.openvulnerability.client.nvd.NvdCveClient._next(NvdCveClient.java:428)
```

Originally reported at

https://github.com/dependency-check/DependencyCheck/issues/7517#issue-2908433719

(With ODC 12.1.0 which uses version 7.2.2 of this library)

--- a/open-vulnerability-clients/src/main/java/io/github/jeremylong/openvulnerability/client/nvd/NvdCveClient.java
+++ b/open-vulnerability-clients/src/main/java/io/github/jeremylong/openvulnerability/client/nvd/NvdCveClient.java
@@ -439,13 +439,17 @@ public class NvdCveClient implements PagedDataSource<DefCveItem> {
                     if (msg != null) {
                         msg = msg.trim();
                         if (msg.contains("Invalid apiKey")) {
+                            String masked;
                             if (this.apiKey.length() > 30) {
-                                String masked = String.format("Invalid API Key: %s-*****-%s",
-                                        this.apiKey.substring(0, 5),
+                                masked = String.format("Invalid API Key: %s-*****-%s", this.apiKey.substring(0, 5),
                                         this.apiKey.substring(this.apiKey.length() - 5, this.apiKey.length()));
-                                throw new NvdApiException(masked);
+                            } else if (this.apiKey.length() < 5) {
+                                masked = String.format(
+                                        "Invalid API Key, length of %d too short to provided a masked partial key",
+                                        this.apiKey.length());
+                            } else {
+                                masked = String.format("Invalid API Key: %s-*****", this.apiKey.substring(0, 5));
                             }
-                            String masked = String.format("Invalid API Key: %s-*****", this.apiKey.substring(0, 5));
                             throw new NvdApiException(masked);
                         } else if (msg.startsWith("resultsPerPage parameter cannot exceed")) {
                             this.resultsPerPage = parseResultsPerPage(msg);

